tests/ciscotest.rb

```diff
@@ -226,6 +226,11 @@ def skip_legacy_defect?(pattern, msg)
     skip(msg) if Utils.image_version?(Regexp.new(pattern))
   end
 
+  def virtual_platform?
```

Comment:
The `@product_id` instance variable already has what you need since it containts the PID from the `show inventory` output so you don't need to query the device with another show command for this case.

```
camden-nx-1# show inventory 
NAME: "Chassis",  DESCR: "NX-OSv Chassis"                        
PID: N9K-NXOSV           ,  VID: V02 ,  SN: 9AK1QGSS07J          
```

Suggest you rewrite the method as follows:

```ruby
def virtual_platform?
  @product_id[/N9K-NXOSV/] ? true : false
end

Reply:
Not really. The PID field is not constant like you mentioned. 
On some platforms, it shows "N9K-NXOSV" like you mentioned, and on some others it is
""N9K-9000v". We cannot depend on this. If you are sure these are the only 2 variations I can do @product_id[/NXOSV|9000v/] ? true : false
but we have to be 100% sure. Can you check with your group and let me know?

Reply:
Hmm... that's unfortunate.  Let me check with Jusheng and get back to you.  I really want to avoid another box query if we can.

Reply:
Ok, I am now checking both NXOSV and 9000v